test/Microsoft.Azure.ServiceBus.UnitTests/TestUtility.cs

```diff
@@ -67,18 +67,27 @@ internal static async Task SendMessagesAsync(IMessageSender messageSender, int m
             Log($"Sent {messageCount} messages");
         }
 
-        internal static async Task<IList<Message>> ReceiveMessagesAsync(IMessageReceiver messageReceiver, int messageCount)
+        internal static async Task<IList<Message>> ReceiveMessagesAsync(IMessageReceiver messageReceiver, int messageCount, TimeSpan timeout = default)
         {
             var receiveAttempts = 0;
             var messagesToReturn = new List<Message>();
+            var stopwatch = Stopwatch.StartNew();
 
-            while (receiveAttempts++ < TestConstants.MaxAttemptsCount && messagesToReturn.Count < messageCount)
+            if (timeout == default)
+            {
+                timeout = TimeSpan.Zero;
+            }
+
+            while (messagesToReturn.Count < messageCount && (receiveAttempts++ < TestConstants.MaxAttemptsCount || stopwatch.Elapsed < timeout))
             {
                 var messages = await messageReceiver.ReceiveAsync(messageCount - messagesToReturn.Count);
-                if (messages != null)
+                if (messages == null)
```

Comment:
This line anyway goes for 60 seconds. Why introduce the stopwatch and timeout which is less than 60 seconds?

Reply:
Despite all messages being sent to the broker and only after that attempted to be received, the broker returns a subset of messages and not all.  `ReceiveAsync(count)`  will not wait for 60 seconds, it will return immediately with whatever number of messages it can get. I think you're confusing this with `ReceiveAsync(count, timeout)` variant.

Reply:
BTW, I had to introduce this change after having random builds failing either on both targets (.NET Core and Framework) or one of those. Sometimes it would all pass. With this [fix](https://github.com/Azure/azure-service-bus-dotnet/pull/621#issuecomment-446853207), the test is [no longer flaky](https://ci.appveyor.com/project/vinaysurya/azure-service-bus-dotnet/builds/20975609) as it was.

Reply:
Yes my bad. I was thinking of something else.

Reply:
All good. If I'd have to count number of times you've corrected me, I'd need more than two hands 😂